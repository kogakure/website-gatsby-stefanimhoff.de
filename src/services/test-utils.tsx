import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';

const customRender = (
  node: React.ReactElement,
  options?: RenderOptions
): RenderResult => {
  const rendered = render(
    <EmojifyProvider>
      <LocalizedDateProvider>{node}</LocalizedDateProvider>
    </EmojifyProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (ui: React.ReactElement, rerenderOptions?: RenderOptions) =>
      customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
