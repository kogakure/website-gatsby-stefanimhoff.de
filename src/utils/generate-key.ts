export const generateKey = (prefix: string) => `${prefix}-${Date.now()}`;
