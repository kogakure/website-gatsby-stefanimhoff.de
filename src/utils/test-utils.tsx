import React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { EmojifyProvider, LocalizedDateProvider } from '../contexts';
import { ThemeType } from '../typings/Theme';
import { lightTheme } from '../themes';

interface CustomTheme {
  theme: ThemeType;
}

type RenderOptionsWithTheme = RenderOptions & CustomTheme;

const customRender = (
  node: React.ReactElement,
  options: RenderOptionsWithTheme = { theme: lightTheme }
): RenderResult => {
  const { theme } = options;
  const rendered = render(
    <ThemeProvider theme={theme}>
      <EmojifyProvider>
        <LocalizedDateProvider>{node}</LocalizedDateProvider>
      </EmojifyProvider>
    </ThemeProvider>,
    options
  );

  return {
    ...rendered,
    rerender: (
      ui: React.ReactElement,
      rerenderOptions?: RenderOptionsWithTheme
    ) => customRender(ui, { container: rendered.container, ...options }),
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
