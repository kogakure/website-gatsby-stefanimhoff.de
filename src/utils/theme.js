/* eslint-disable */
(function() {
  const root = document.getElementsByTagName('html')[0];
  window.__onThemeChange = function() {};

  function setTheme(newTheme) {
    window.__theme = newTheme;
    preferredTheme = newTheme;
    currentTheme = newTheme === 'light' ? 'dark' : 'light';
    root.classList.add(newTheme);
    root.classList.remove(currentTheme);
    window.__onThemeChange(newTheme);
  }

  let preferredTheme;

  try {
    preferredTheme = localStorage.getItem('theme');
  } catch (err) {}

  window.__setPreferredTheme = function(newTheme) {
    setTheme(newTheme);
    try {
      localStorage.setItem('theme', newTheme);
    } catch (err) {}
  };

  const darkQuery = window.matchMedia('(prefers-color-scheme: dark)');

  darkQuery.addListener(function(e) {
    window.__setPreferredTheme(e.matches ? 'dark' : 'light');
  });

  setTheme(preferredTheme || (darkQuery.matches ? 'dark' : 'light'));
})();
