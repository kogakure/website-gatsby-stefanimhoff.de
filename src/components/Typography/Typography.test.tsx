import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../utils/test-utils';

import { Title, Headline, Subline, Paragraph } from '.';

describe('Typography', () => {
  describe('Title', () => {
    test('renders correctly', () => {
      const { container } = render(<Title>Title</Title>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Headline', () => {
    test('renders correctly', () => {
      const { container } = render(<Headline>Headline</Headline>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Subline', () => {
    test('renders correctly', () => {
      const { container } = render(<Subline>Subline</Subline>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Paragraph', () => {
    test('renders correctly', () => {
      const { container } = render(<Paragraph>Paragraph</Paragraph>);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
