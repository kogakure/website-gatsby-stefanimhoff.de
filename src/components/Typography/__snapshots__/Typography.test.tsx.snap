// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Headline renders correctly 1`] = `
.c0 {
  font-size: 3.272rem;
  font-weight: 900;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.0;
  margin-bottom: 2.074rem;
  margin-top: 0;
}

<h2
  class="c0"
  font-size="5"
  font-weight="900"
  letter-spacing="normal"
>
  Headline
</h2>
`;

exports[`Typography Paragraph renders correctly 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.6;
  margin-top: 0;
  margin-bottom: 2.488rem;
}

<p
  class="c0"
  font-size="3"
  font-weight="400"
  letter-spacing="normal"
>
  Paragraph
</p>
`;

exports[`Typography Subline renders correctly 1`] = `
.c0 {
  font-size: 2.023rem;
  font-weight: 900;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.0;
  margin-bottom: 1.728rem;
  margin-top: 0;
}

<h3
  class="c0"
  font-size="4"
  font-weight="900"
  letter-spacing="normal"
>
  Subline
</h3>
`;

exports[`Typography Title renders correctly 1`] = `
.c0 {
  font-size: 5.295rem;
  font-weight: 900;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1.0;
  margin-bottom: 2.986rem;
  margin-top: 0.694rem;
}

<h1
  class="c0"
  font-size="6"
  font-weight="900"
  letter-spacing="normal"
>
  Title
</h1>
`;
